lang/c++/impl/NodeImpl.cc

```diff
@@ -318,11 +319,33 @@ void NodePrimitive::printDefaultToJson(const GenericDatum &g, std::ostream &os,
       os << g.value<int64_t>();
       break;
     case AVRO_FLOAT:
-      os << g.value<float>();
-      break;
+      {
+          auto val = g.value<float>();
+          if (boost::math::isnan(val)) {
+              os << "\"NaN\"";
+          } else if (val == std::numeric_limits<float>::infinity()) {
+              os << "\"Infinity\"";
+          } else if (val == -std::numeric_limits<float>::infinity()) {
```

Comment:
This might run into problems as we are comparing floating point numbers? Probably we want to check this with
`std::isinf(negative_infinity) && negative_infinity < std::numeric_limits<float>::lowest()`. 
Same for positive infinity
`std::isinf(positive_infinity) && positive_infinity > std::numeric_limits<float>::lowest()`

Reply:
i believe there is only one binary representation possible for positive and negative infinity. given that, i would assume straight comparison is ok?